components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/security/handler/AuthorizationHandler.java

```diff
@@ -0,0 +1,123 @@
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.micro.integrator.management.apis.security.handler;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.MessageContext;
+import org.wso2.micro.core.util.CarbonException;
+import org.wso2.micro.integrator.management.apis.Constants;
+import org.wso2.micro.integrator.management.apis.ManagementApiUndefinedException;
+import org.wso2.micro.integrator.security.MicroIntegratorSecurityUtils;
+import org.wso2.micro.integrator.security.user.api.UserStoreException;
+
+import javax.xml.stream.XMLStreamException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import static org.wso2.micro.integrator.management.apis.Constants.USERNAME_PROPERTY;
+
+/**
+ * Handler to be used for resources that required admin privileges.
+ * <p>
+ * This handler will only authorize requests with a token that belong to an admin user.
+ */
+public class AuthorizationHandler extends SecurityHandlerAdapter {
+
+    private static final Log LOG = LogFactory.getLog(AuthorizationHandler.class);
+    private String name;
+
+    public AuthorizationHandler(String context) throws CarbonException, XMLStreamException, IOException,
+            ManagementApiUndefinedException {
+        super(context);
+    }
+
+    @Override
+    protected void populateDefaultResources() {
+        defaultResources = new ArrayList<>(1);
+        defaultResources.add(Constants.PREFIX_USERS);
+    }
+
+    @Override
+    public Boolean invoke(MessageContext messageContext) {
+
+        this.messageContext = messageContext;
+        return super.invoke(messageContext);
+    }
+
+    @Override
+    public String getName() {
+
+        return this.name;
+    }
+
+    @Override
+    public void setName(String name) {
+
+        this.name = name;
+    }
+
+    @Override
+    protected Boolean authenticate(String authHeaderToken) {
+        if (useCarbonUserStore) {
+            //Uses carbon user store
+            try {
+                return processAuthorizationWithCarbonUserStore();
+            } catch (UserStoreException e) {
+                LOG.error("Error while authenticating with carbon user store", e);
+                return false;
+            }
+        } else {
+            //Uses in memory user store
+            LOG.error("Authorization is not supported with the in memory user store. The request will be authorized. "
+                      + "Please plug in a user store for the correct functionality");
```

Comment:
This should be , please plugin in a JDBC user store ?

Reply:
done